src/test_helper.rs

```diff
@@ -0,0 +1,188 @@
+
+        app.get_terminal()
+            .backend_mut()
+            .resize(buffer_size.width, buffer_size.height);
+
+        app.run()
+    });
+
+    let event_tx = event_rx.blocking_recv().unwrap();
+
+    let buffer = message_rx.blocking_recv().unwrap();
+
+    event_tx.blocking_send(Event::Quit).unwrap();
+    let _ = join_handle.join().unwrap();
+    buffer
+}
+
+/// Render a widget and return the terminal to check the generated output
+///
+/// * `buffer_size` - The terminal output buffer is set to that size.
+/// * `sut` - (system under test) The widget to render.
+pub fn render_widget(buffer_size: Size, sut: &mut impl Widget) -> Terminal<TestBackend> {
+    let mut messages = vec![];
+    let mut cx_state = CxState::new(&mut messages);
+    let mut widget_state = WidgetState::new();
+
+    let mut layout_cx = LayoutCx {
+        cx_state: &mut cx_state,
+        widget_state: &mut widget_state,
+    };
+    let backend = TestBackend::new(buffer_size.width, buffer_size.height);
+
+    let mut terminal = Terminal::new(backend).unwrap();
+    let _size = sut.layout(
+        &mut layout_cx,
+        &BoxConstraints::new(
+            kurbo::Size::ZERO,
+            kurbo::Size {
+                width: buffer_size.width.into(),
+                height: buffer_size.height.into(),
+            },
+        ),
+    );
+
+    let mut paint_cx = PaintCx {
+        cx_state: &mut cx_state,
+        widget_state: &mut widget_state,
+        terminal: &mut terminal,
+        override_style: Style::default(),
+    };
+
+    sut.paint(&mut paint_cx);
+    terminal.flush().unwrap();
+
+    terminal
+}
+
+/// This widget provides access to the terminal output of its children
+///
+/// After its children were painted it calls the flush() and clones the
+/// terminal's buffer and sends it using the passed [Sender<Buffer>].
+///
+/// This is handy for snapshot tests.
+pub struct DebugView<V, T, A> {
+    content: V,
+    debug_chan_tx: mpsc::Sender<Buffer>,
+    phantom: PhantomData<fn() -> (T, A)>,
+}
+
+pub fn debug_view<V, T, A>(content: V, debug_chan_tx: mpsc::Sender<Buffer>) -> DebugView<V, T, A> {
+    DebugView {
+        content,
+        debug_chan_tx,
+        phantom: PhantomData,
+    }
+}
+
+impl<T, A, V> ViewMarker for DebugView<V, T, A> {}
+
+impl<T, A, V: View<T, A>> View<T, A> for DebugView<V, T, A> {
+    type State = V::State;
+
+    type Element = DebugWidget;
+
+    fn build(&self, cx: &mut Cx) -> (xilem_core::Id, Self::State, Self::Element) {
+        let (id, state, element) = self.content.build(cx);
+        (
+            id,
+            state,
+            DebugWidget::new(element, self.debug_chan_tx.clone()),
+        )
+    }
+
+    fn rebuild(
+        &self,
+        _cx: &mut Cx,
+        _prev: &Self,
+        _id: &mut xilem_core::Id,
+        _state: &mut Self::State,
+        _element: &mut Self::Element,
+    ) -> crate::widget::ChangeFlags {
+        ChangeFlags::empty()
```

Comment:
Well the child view should still be rebuilt, otherwise no updates can be done to this view, I just meant that the extra variable was unnecessary.

```suggestion
        cx: &mut Cx,
        prev: &Self,
        id: &mut xilem_core::Id,
        state: &mut Self::State,
        element: &mut Self::Element,
    ) -> ChangeFlags {
        let element = element
            .content
            .downcast_mut()
            .expect("The DebugView content widget changed its type, this should never happen!");
        self.content.rebuild(cx, &prev.content, id, state, element)
```